StudyHub/StudyHub.BLL/Services/Assignments/AssignmentTaskService.cs

```diff
@@ -63,6 +63,11 @@ public async Task<List<AssignmentTaskDTO>> GetAssignmentTaskAsync(Guid assignmen
 
     public async Task<AssignmentTaskDTO> UpdateAssignmentTaskAsync(Guid assignmentTaskId, UpdateAssignmentTaskDTO assignmentTask)
     {
+        var assignment = await _assignmentRepository.SingleAsync(assignment => assignment.Tasks.FirstOrDefault(task => task.Id == assignmentTaskId)!.Id == assignmentTaskId);
```

Comment:
Why u use SingleAsync?

Reply:
I think whole search is incorrect, discussed with Ivan, he will check